src/main/scala/sbtghactions/WorkflowJob.scala

```diff
@@ -16,6 +16,16 @@
 
 package sbtghactions
 
+sealed trait WorkflowJobBase
+
+final case class ReusableWorkflowJob(
+  id: String,
+  name: String,
+  uses: WorkflowRef,
+  cond: Option[String] = None,
+  needs: List[String] = List(),
```

Comment:
Should we also add `javas` and `scalas` here (like with `WorkflowJob`) so that these values can be accessable if someone wants to create a custom workflow that uses those values?

Reply:
I am not so sure about that. I think you can make a very good argument, that it is after this PR more or less a smell, that `WorkflowJob` does not just represent the GH api and that as a pure data structure should only do that.
And yes, I bring this argument, because I think it is the better design. What I did not think about yet is, if this would bing migration issues.
But I think, in a perfect world there would be no `oses`, `javas` and `scalas` here, but the plugin would build the right `WorkflowJob` out of the sbt keys before rendering it.

Reply:
After thinking about it, I think it does make sense in a future PR because as you said some things might need to be moved around and this PR has already been open for some time. The reason I brought this up is because this is actually a usecase for one of the projects I maintain, i.e. we definitely have a use for custom workflows however we are still going to be running jobs on JVM's defined by `githubWorkflowJavaVersions` and OS's defined by `githubWorkflowOSes` (amongst other things) and without such a change you basically have to redefine the exact same values (once in `githubWorkflowJavaVersions` and another in your custom workflow) which also has the obvious risk of them accidentally going out of sync.

Reply:
You actually don't. You will define the workflow anyway in an sbt file or your own sbt plugin. And there you can know the key. You can just use it somewhat like this.

```scala
  val SetupScala: WorkflowStep.Use = WorkflowStep.Use(
    name = Some("Setup Java and Scala"),
    ref = UseRef.Public("coursier", "setup-action", "v1"),
    params = Map("jvm" -> GenerativeKeys.githubWorkflowJavaVersions.value.head.version),
    )
```

What I could understand is, having a method on the data types like "withJavaVersions" that sets a matrix up in the right way. But I would really remove every data from the structure that is not a reflection of the github api.

Reply:
> If you like, we could try to have a call and try to talk about a strategy for this or pairing on a follow up issue.

Sure! Next week I will have some time (feel free to email me to discuss, its on my profile)

For now I will merge the PR in a few days to see if we can get any more additional reviews.